### PR TITLE
Season switch : First step 

### PR DESCRIPTION
**Pull request by emcoding**
## To do's left
- [x] Routing 
- [x] Changing if-lsesleselsle into `case`
- [x] Change :switch_seasons to :switch_season_phase
- [x] Update Readme 

left overs:

To do
- [x] Add Optimize comment https://github.com/rails-girls-summer-of-code/rgsoc-teams/pull/304#discussion-diff-38808635R60
- [x] Change env into unless Rails.env.production?

### CODE REVIEW

**Comment by emcoding** [comment]
Before the next steps, I like to know if I am on the right path.

(@carpodaster )

**Comment by carpodaster** [comment]
Orga::Seasons is a good place to fiddle with the seasons for development reasons I think :+1: 

Changing to a concrete Season like that would probably work for the current request, but not for all subsequent ones (the value of  `@season` will be gone after the request has ended). As mentioned in rails-girls-summer-of-code/rgsoc-teams#301, changing the season in development mode probably requires _editing_ the timestamps for the _current_ season.

A more elaborate, albeit more complicated approach would be to allow attach a season record to a user (the user being a developer). With that, the current_season helper can either grab the one defined by the developer or get `Season.current`.

This would only work where we access the current_season helper, ie. one has to make sure we indeed use it wherever possible. Some portions of the code (mailings, for instance) don't have access to that helper and won't be affected by a developer's season switch.

**Comment by emcoding** [comment]
O dear, I misunderstood what you meant in #301. Sorry. Thanks again.

File: app/views/layouts/_header.html.slim [hunk 1]
```diff
@@ -14,6 +14,7 @@ header
       li = link_to 'Community', users_path, class: params[:controller] == 'users' ? 'active' : ''
       - if current_season.applications_open?
         li = link_to 'Conferences',  conferences_path,  class: params[:controller] == 'conferences' ? 'active' : ''
+      - if current_season.started?
         li = link_to 'Jobs',   job_offers_path,   class: params[:controller] == 'job_offers' ? 'active' : ''
       /- if signed_in? && current_user.roles.includes?('reviewer')  # TODO bring this into a reviewer dashboard
```
Comment (emcoding) [inline app/views/layouts/_header.html.slim:19]: The JobOffers feature and so the link to Jobs will be deleted before the end of SoC

**Comment by emcoding** [comment]
@carpodaster I left the Switch in the Orga NS. This set up is WIP, only to test how to 'make it work'.
In this setup the current season's date are set, and that changes the Season.current dates. Not sure if that is exactly what you DIDN't want to happen.

(I still have to discuss the code itself with the coaches.)

File: config/routes.rb [hunk 2]
```diff
@@ -71,9 +71,13 @@
 
   namespace :orga do
     resources :teams
-    resources :seasons
+    resources :seasons do
+      member do
+        get :switch_seasons
```
Comment (carpodaster) [inline config/routes.rb]: From a RESTful perspective, it's more a PUT – but that's nit-picking so don't mind me :wink:

Comment (yorickpeterse) [inline config/routes.rb]: If it's meant to update something then _technically_ it should be PATCH :wink:

Comment (emcoding) [inline config/routes.rb]: Nice! On it.

**Comment by emcoding** [comment]
Now the logic is in the model and the season settings are better secured. Thanks, André.

**Comment by emcoding** [comment]
EDIT @yorick explained the mistake, so this is on hold.

I did not figure out yet how to change the `get`-route into `put` or `patch`  ( @YorickPeterse )
- Error: Missing template seasons_switch
- Browser output:

```
Started PATCH "/orga/seasons/switch_seasons" for ::1 at 2015-09-02 21:31:40 +0200
Processing by Orga::SeasonsController#switch_seasons as HTML
  Parameters: {"utf8"=>"✓", "authenticity_token"=>"dHZcXDAtdhNr5qpAj9L5PPmOmhecx3z+7BAR/4VAv0WMha0Dbtn+fvx6BFMK/m6Jlx/rJralFNGFLl2i1J5Tgg==", "option"=>"Application", "commit"=>"Switch phase"}
  User Load (0.3ms)  SELECT  "users".* FROM "users" WHERE "users"."id" = $1  ORDER BY "users"."id" ASC LIMIT 1  [["id", 31]]
   (0.2ms)  SELECT COUNT(*) FROM "roles" WHERE "roles"."user_id" = $1 AND "roles"."name" IN ('organizer', 'developer')  [["user_id", 31]]
Completed 500 Internal Server Error in 15ms (ActiveRecord: 0.5ms)
[Rollbar] Reporting exception: Missing template orga/seasons/switch_seasons, orga/base/switch_seasons, application/switch_seasons with {:locale=>[:en], :formats=>[:html], :variants=>[], :handlers=>[:erb, :builder, :raw, :ruby, :slim, :coffee]}. Searched in:
  * "/Users/maud/Desktop/code/soc/app/views"
  * "/Users/maud/.rvm/gems/ruby-2.2.2/gems
... etc
```

**Comment by emcoding** [comment]
Sorry, I surrender. I can't properly change the get-route into a put/patch. It is on my list for when I have a bit of time left in a live coaching session. 
But as it was classified as 'nit-picking' by all of you, I think it shouldn't stand in the way of shipping. Right?
So, please, your verdict!

**Comment by emcoding** [comment]
@nathanvda Thanks for the :collection. I tried it, not sure what I did wrong, but I will remember this.

I agree, :switch_seasons is not a good name. I'll change that.
And +1 on the @fakeseason name also. Changed that.

**Comment by emcoding** [comment]
Tests pass.

File: config/routes.rb [hunk 3]
```diff
@@ -71,9 +71,13 @@
 
   namespace :orga do
     resources :teams
-    resources :seasons
+    resources :seasons do
+      member do
```
Comment (andremedeiros) [inline config/routes.rb]: Can we indent this correctly?

**Comment by emcoding** [comment]
Thanks everybody!! I am sending it to upstream.

**Pull request merged**